File: app/src/main/java/com/ericho/coupleshare/act/TestCollapseToolbarLayoutAndTabLayoutAct.java
package com.ericho.coupleshare.act;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.design.widget.TabLayout;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;
import android.widget.ImageView;

import com.ericho.coupleshare.R;
import com.ericho.coupleshare.adapter.MyFragPageAdapter;
import com.ericho.coupleshare.frag.DummyFrag;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by steve_000 on 9/5/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.act
 */

public class TestCollapseToolbarLayoutAndTabLayoutAct extends RxLifecycleAct {
    public static final String FLAG = "TestCollapseToolbarLayoutAndTabLayoutAct";

    @BindView(R.id.toolbar_layout)
    CollapsingToolbarLayout collapsingToolbarLayout;
    @BindView(R.id.viewPager)
    ViewPager viewPager;
    @BindView(R.id.tabLayout)
    TabLayout tabLayout;
    @BindView(R.id.img)
    ImageView img;
    @BindView(R.id.toolbar)
    Toolbar toolbar;

    FragmentPagerAdapter fragmentPagerAdapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.act_test_collapsing_toolbar_tablayout);
        ButterKnife.bind(this);
        init();
    }

    private void init() {
        fragmentPagerAdapter = new MyFragPageAdapter(getSupportFragmentManager());
        viewPager.setAdapter(fragmentPagerAdapter);
        tabLayout.setupWithViewPager(viewPager);
    }
}

File: app/src/main/java/com/ericho/coupleshare/act/TestCollapseToolbarLayoutAndTabLayoutAct.kt
package com.ericho.coupleshare.act

import android.os.Bundle
import android.support.design.widget.CollapsingToolbarLayout
import android.support.v4.app.FragmentPagerAdapter
import butterknife.bindView
import com.ericho.coupleshare.R
import com.ericho.coupleshare.adapter.MyFragPageAdapter
import kotlinx.android.synthetic.main.act_test_collapsing_toolbar_tablayout.*

/**
 * Created by steve_000 on 8/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.act
 */
class TestCollapseToolbarLayoutAndTabLayoutAct:RxLifecycleAct() {


    val collapsingToolbarLayout: CollapsingToolbarLayout by bindView(R.id.toolbar_layout)



    internal var fragmentPagerAdapter: FragmentPagerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.act_test_collapsing_toolbar_tablayout)
        init()
    }

    private fun init() {
        fragmentPagerAdapter = MyFragPageAdapter(supportFragmentManager)
        viewPager.setAdapter(fragmentPagerAdapter)
        tabLayout.setupWithViewPager(viewPager)
    }
    companion object {
        @JvmField
        val FLAG = "TestCollapseToolbarLayoutAndTabLayoutAct"
    }
}